### ISSUE TITLE
Global suppression with SuppressMessage doesn't work with explicit interface implementations.

### ISSUE THREAD

**Issue by alexperovich** [dotnet/roslyn#15728]
**Version Used**: 
Microsoft (R) Visual C# Compiler version 2.0.0.61104
Copyright (C) Microsoft Corporation. All rights reserved.

**Steps to Reproduce**:

1. Compile this gist https://gist.github.com/alexperovich/c18cda6e63c105702a0d8ef07e8ee729 with the fxcop analyzers enabled. This will report CA2002 on line 18.
2. Add the following suppress message attributes to the file

```csharp
[assembly: System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Reliability", "CA2002:DoNotLockOnObjectsWithWeakIdentity", Scope = "member", Target = "ConsoleApp1.Program.#ConsoleApp1.IBar.Foo()")]
[assembly: System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Reliability", "CA2002:DoNotLockOnObjectsWithWeakIdentity", Scope = "member", Target = "M:ConsoleApp1.Program.IBar#Foo()")]
```

3. Recompile

**Expected Behavior**:

The file compiles without warnings.

**Actual Behavior**:

CA2002 is still reported. It appears that SuppressMessage.Target doesn't work with explicit interface implementations.
The first suppression is the FxCop form generated by visual studio. The second is the documentation comment id.

**Comment by mavasani** [comment]
@alexperovich Seems like the Roslyn suppress message attribute decode doesn't understand the FXCop specific suppression syntax. Can you try adding the below suppression as well:

```
[assembly: System.Diagnostics.CodeAnalysis.SuppressMessage("Reliability", "CA2002:Do not lock on objects with weak identity", Justification = "<Pending>", Scope = "member", Target = "~M:ConsoleApp1.Program.ConsoleApp1#IBar#Foo")]
```
